src/SDK/Trace/Behavior/LoggerAwareTrait.php

```diff
@@ -23,20 +27,65 @@ public function setDefaultLogLevel(string $logLevel): void
         $this->defaultLogLevel = $logLevel;
     }
 
+    /**
+     * @return static
+     */
+    public function withLogger(LoggerInterface $logger)
+    {
+        $this->logger = $logger;
+
+        return $this;
+    }
+
     /**
      * @param string $message
      * @param array $context
      * @param string|null $level
      */
     protected function log(string $message, array $context = [], ?string $level = null): void
     {
+        //TODO basic info on which class created the message, cheaper than calculating from a stack trace
+        $context['source'] = __CLASS__;
         $this->getLogger()->log(
             $level ?? $this->defaultLogLevel,
             $message,
             $context
         );
     }
 
+    protected function debug(string $message, array $context = []): void
+    {
+        $this->log($message, $context, LogLevel::DEBUG);
+    }
+
+    protected function warning(string $message, array $context = []): void
+    {
+        $this->log($message, $context, LogLevel::WARNING);
+    }
+
+    protected function error(string $message, array $context = []): void
+    {
+        $this->log($message, $context, LogLevel::ERROR);
+    }
+
+    protected function info(string $message, array $context = []): void
```

Comment:
I think there are a some [severity levels](https://github.com/open-telemetry/opentelemetry-php/blob/main/src/SDK/Logs/PsrSeverityMapperInterface.php) missing.

Reply:
I added NOTICE, but I feel like critical and higher should not be needed...? (or, can be added if we do have a use-case)

Reply:
Ok.